### PR TITLE
Solves 494: added destructor to ssm connection plugin

### LINKED ISSUE #494: SSM connection plugin doesnt properly close connections
##### SUMMARY
When trying to run a big playbook using the SSM connection plugin, it randomly hangs in the middle of it. Very rarely am I able to run the entire playbook without issues.

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME
ssm connection plugin

##### ANSIBLE VERSION
```
ansible 2.10.5
  config file = /Users/xxx/.ansible.cfg
  configured module search path = ['/Users/xxx/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.8/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.8.8 (default, Feb 21 2021, 10:35:39) [Clang 12.0.0 (clang-1200.0.32.29)]
```

##### CONFIGURATION
```
DEFAULT_HOST_LIST(env: ANSIBLE_INVENTORY) = ['/Users/xxx/ansible_hosts']
DEFAULT_VAULT_IDENTITY_LIST(/Users/xxx/.ansible.cfg) = ['xx@~/.vault_pass.txt', 'xxx@~/.vault_pass_xxx.txt', 'xxx@~/.vault_pass_xxx.txt']
```

Ansible variables used in the playbook for configuring the SSM plugin:
```
  vars:
    ansible_connection: community.aws.aws_ssm
    ansible_aws_ssm_bucket_name: xxx-ansible-ssm
    ansible_aws_ssm_region: eu-west-1
```

##### OS / ENVIRONMENT
Target OS: `Amazon-Linux 2`

##### STEPS TO REPRODUCE
I dont have exact steps to replicate this issue, it seems to happen to bigger playbooks. And happens randomly, sometimes it dies immediately, sometimes it dies in the middle or end, and very rarely does it complete without issues.

##### EXPECTED RESULTS
To complete the playbook without hanging.

##### ACTUAL RESULTS
When running in verbose mode, these are the last lines printed, i left the playbook running for 10 minutes and no change happened after which i stopped it manually:
```
....

<i-xxx> ESTABLISH SSM CONNECTION TO: i-xxx
<i-xxx> SSM CONNECTION ID: xxx-0a55f9c52a37613a0
<i-xxx> EXEC echo ~
^C [ERROR]: User interrupted execution
```

If I SSH to the server, it seems there are a lot of connections left hanging, this is the output of `ps -e --forest -o ppid,pid,user,command`:
output of ps command

This has been an issue for me for several releases of the ssm connection plugin.

### PR DESCRIPTION
**Pull request by Chilinot**
##### SUMMARY
Fixes #494

Adds destructor to the SSM connection plugin to ensure connections are properly cleaned up after usage.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
SSM connection plugin

### CODE REVIEW

**Review by tremble — CHANGES_REQUESTED** [review]
Hi @Chilinot,

Thanks for taking the time to dig into the issue and open this PR.  If possible please could you add a changelog fragment: https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to

The change itself looks sensible.

**Comment by tremble** [comment]
@hgrgic you reported also experiencing the issue on #494 if possible please could you test the change in this PR and see if it fixes the issue you're seeing.

**Comment by hgrgic** [comment]
Hi @tremble  and @Chilinot. Thanks for this update. 

I have tested the proposed fix with my playbooks. Immediately after adding the proposed change I noticed that sessions were terminated properly. At the same time, looking in the AWS Session Manager console, under sessions, at most time I saw only 1 connection being open.

I have repeated the test with the same playbook but before repeating I have removed the proposed change. In this instance I saw more than 20 connection remaining open even after the execution of playbooks has finished. This tells me that the fix is indeed effective.

I recommend merging this PR as soon as possible!

**Comment by Chilinot** [comment]
@tremble I have now added a changelog fragment as required.

File: changelogs/fragments/542-ensure-ssm-plugin-terminates-connections.yml [hunk 1]
```diff
@@ -0,0 +1,2 @@
+bug_fixes:
+  - aws_ssm - Adds destructor to SSM connection plugin to ensure connections are properly cleaned up after usage.
```
Comment (tremble) [inline changelogs/fragments/542-ensure-ssm-plugin-terminates-connections.yml:2]:
```suggestion
  - aws_ssm - Adds destructor to SSM connection plugin to ensure connections are properly cleaned up after usage (https://github.com/ansible-collections/community.aws/pull/542).
```

**Review by tremble — APPROVED** [review]
LGTM - I'll let the test suites do their thing and hopefully we can get this merged.

File: changelogs/fragments/542-ensure-ssm-plugin-terminates-connections.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+bug_fixes:
```
Comment (tremble) [inline changelogs/fragments/542-ensure-ssm-plugin-terminates-connections.yml:1]:
```suggestion
bugfixes:
```

**Comment by tremble** [comment]
Tests didn't run in zuul for some reason.  Tested locally and they still pass.

**Comment by hgrgic** [comment]
@tremble What is usually the expected cadence for this merged fix to be included in a release?

**Comment by tremble** [comment]
Our last release was almost 2 months ago, I'd expect 1.5.0 to land 'soon'.

**Pull request merged**